src/game/engine/cell.cpp

```diff
@@ -994,17 +995,15 @@ void CellClass::Wall_Update()
 
     static FacingType _offsets[] = { FACING_NORTH, FACING_EAST, FACING_SOUTH, FACING_WEST, FACING_NONE };
 
-    for (int index = 0; index < ARRAY_SIZE(_offsets); ++index) {
+    for (int index = 0; index <= ARRAY_SIZE(_offsets); ++index) {
```

Comment:
Why was this changed? This will cause it to go out of bounds if it iterates over the entire array. Array size is 1 based, but iteration is 0 based.